app/src/main/java/org/simple/clinic/patient/recent/RecentPatientItem.kt

```diff
@@ -23,37 +26,56 @@ data class RecentPatientItem(
     val age: Int,
     val lastBp: LastBp?,
     val gender: Gender
-) : RecentPatientItemType(uuid.hashCode().toLong()) {
+) : RecentPatientItemType<RecentPatientItem.RecentPatientViewHolder>(uuid.hashCode().toLong()) {
 
   override fun getLayout(): Int = R.layout.recent_patient_item_view
 
-  @SuppressLint("SetTextI18n")
-  override fun bind(viewHolder: ViewHolder, position: Int) {
-    viewHolder.itemView.apply {
-      recentpatient_item_title.text = "$name, $age"
-
-      val lastBpText = lastBpText(context)
-      recentpatient_item_last_bp.text = lastBpText
-      recentpatient_item_last_bp_label.visibleOrGone(lastBpText.isNotBlank())
-      recentpatient_item_gender.setImageResource(gender.displayIconRes)
+  override fun createViewHolder(itemView: View): RecentPatientViewHolder {
+    return RecentPatientViewHolder(itemView)
+  }
 
-      setOnClickListener {
-        uiEvents.onNext(RecentPatientItemClicked(patientUuid = uuid))
-      }
+  @SuppressLint("SetTextI18n")
+  override fun bind(viewHolder: RecentPatientViewHolder, position: Int) {
+    viewHolder.itemView.setOnClickListener {
+      uiEvents.onNext(RecentPatientItemClicked(patientUuid = uuid))
     }
-  }
 
-  private fun lastBpText(context: Context): String =
-      lastBp?.run { "$systolic/$diastolic, ${updatedAtRelativeTimestamp.displayText(context)}" } ?: ""
+    viewHolder.render(name, age, lastBp, gender)
+  }
 
   data class LastBp(
       val systolic: Int,
       val diastolic: Int,
       val updatedAtRelativeTimestamp: RelativeTimestamp
   )
+
+  class RecentPatientViewHolder(rootView: View) : ViewHolder(rootView) {
+    private val nameAgeTextView by bindView<TextView>(R.id.recentpatient_item_nameage)
+    private val lastBpTextView by bindView<TextView>(R.id.recentpatient_item_last_bp)
+    private val lastBpLabel by bindView<View>(R.id.recentpatient_item_last_bp_label)
+    private val genderImageView by bindView<ImageView>(R.id.recentpatient_item_gender)
+
+    fun render(
+        name: String,
+        age: Int,
+        lastBp: LastBp?,
+        gender: Gender
+    ) {
+      nameAgeTextView.text = itemView.resources.getString(R.string.patients_recentpatients_nameage, name, age)
+
+      val lastBpText = lastBpText(itemView.context, lastBp)
+      lastBpTextView.text = lastBpText
+      lastBpLabel.visibleOrGone(lastBpText.isNotBlank())
+      genderImageView.setImageResource(gender.displayIconRes)
+    }
+
+    private fun lastBpText(context: Context, lastBp: LastBp?): String {
+      return lastBp?.run { "$systolic/$diastolic, ${updatedAtRelativeTimestamp.displayText(context)}" } ?: ""
+    }
+  }
 }
 
-object SeeAllItem : RecentPatientItemType(0) {
+object SeeAllItem : RecentPatientItemType<ViewHolder>(0) {
```

Comment:
Can you fix the `0` being sent as the `adapterId`?
Can you create a constant instead and pass that?

Reply:
I want to consider how we generate adapter IDs across the board since we don't do it very well at the moment. I'll do this when we get around to that?

Reply:
Sure.